src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calcAvgGrades(grades: Array<number>):number {
+  const sum: number = grades.reduce((acc:number, curr:number) => acc + curr, 0);
+  const avg: number = (sum / grades.length);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return avg;
+}
+
+export function sortStudents(
+  students: Array<Student>,
+  sortBy: SortType,
+  order: SortOrder,
+): Array<Student> {
+  const studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      studentsCopy.sort((firstStudent, secondStudent) => (order === 'asc'
+        ? firstStudent[sortBy].localeCompare(secondStudent[sortBy])
+        : secondStudent[sortBy].localeCompare(firstStudent[sortBy])
+      ));
+      break;
+    case SortType.Age:
+    case SortType.Married:
+      studentsCopy.sort((firstStudent, secondStudent) => (order === 'asc'
+        ? +firstStudent[sortBy] - +secondStudent[sortBy]
```

Comment:
```suggestion
        ? Number(firstStudent[sortBy]) - Number(secondStudent[sortBy])
```
Number() is more obvious way